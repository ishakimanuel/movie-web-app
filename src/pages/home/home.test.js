const { render, screen } = require('@testing-library/react');
const { MemoryRouter } = require('react-router-dom');
const { default: HomePage } = require('./home.page');

function renderHomePageWithRouter() {
  return render(<HomePage />, { wrapper: MemoryRouter });
}

describe('Home Page', () => {
  test('Should render consistent homepage UI', () => {
    const homepage = renderHomePageWithRouter();

    const homepageJson = homepage.asFragment();
    expect(homepageJson).toMatchSnapshot();
  });

  test('should render Logo', () => {
    renderHomePageWithRouter();
    const logoContainer = screen.getByRole('heading', { level: 1 });
    const movieText = screen.getByText(/movie/i);
    const appText = screen.getByText(/app/i);
    expect(logoContainer).toBeInTheDocument();
    expect(movieText).toBeInTheDocument();
    expect(appText).toBeInTheDocument();
  });

  test('should render exlore button', () => {
    renderHomePageWithRouter();
    const buttonExplore = screen.getByRole('link', {
      name: /explore-button/i,
    });
    expect(buttonExplore).toBeInTheDocument();
    expect(buttonExplore).toHaveTextContent('Explore Now');
  });
});
