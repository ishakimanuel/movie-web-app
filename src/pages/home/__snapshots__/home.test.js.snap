// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Should render consistent homepage UI 1`] = `
<DocumentFragment>
  <div
    class="homepage flex h-screen bg-dark flex-col items-center justify-center text-center"
  >
    <div
      class="w-50"
    >
      <div>
        <h1
          class="logo ff-heading text-8xl undefined"
        >
          <span
            class="text-primary"
          >
            Movie
          </span>
          <span
            class="text-white"
          >
            App
          </span>
        </h1>
      </div>
      <a
        aria-label="explore-button"
        class="button font-bold text-white uppercase ff-heading p-4 text-2xl bg-primary mt-5 inline-block"
        href="/movies"
      >
        Explore Now
      </a>
    </div>
  </div>
</DocumentFragment>
`;
